ci/scripts/python_wheel_windows_test.bat

```diff
@@ -56,8 +56,15 @@ python -c "import pyarrow.dataset" || exit /B 1
 python -c "import pyarrow.flight" || exit /B 1
 python -c "import pyarrow.fs" || exit /B 1
 python -c "import pyarrow.json" || exit /B 1
+python -c "import pyarrow.orc" || exit /B 1
 python -c "import pyarrow.parquet" || exit /B 1
 python -c "import pyarrow.substrait" || exit /B 1
 
+@rem Download IANA Timezone Database for ORC C++
```

Comment:
So ORC needs this unconditionally? Why is that?

Reply:
The root cause is that when creating an ORC reader it tries to read the local timezone: https://github.com/apache/orc/blob/main/c%2B%2B/src/Reader.cc#L251 which by default searchs `/usr/share/zoneinfo` if env `TZDIR` is unset: https://github.com/apache/orc/blob/main/c%2B%2B/src/Timezone.cc#L659

So the workaround here is to download tzdb and set env `TZDIR` to make the test happy. I think we can improve this on the ORC side:
- More search path candidates for IANA TZDB.
- Read the TZDB lazily as the metadata field is optional and for timestamp type only: https://github.com/apache/orc-format/blob/main/src/main/proto/orc/proto/orc_proto.proto#L181
- Ship TZDB if the build process happens on Windows.